sdks/go/pkg/beam/core/runtime/exec/translate.go

```diff
@@ -34,9 +34,13 @@ import (
 	"github.com/golang/protobuf/ptypes"
 )
 
+// TODO(lostluck): 2018/05/28 Extract these from the canonical enums in beam_runner_api.proto
```

Comment:
I have a code snippet that does most of this. Let me flesh it out and send it to you.